lua/barbar/layout.lua

```diff
@@ -42,31 +57,57 @@ local Layout = { buffers = {} }
 --- Calculate the current layout of the bufferline.
 --- @return barbar.layout.data
 function Layout.calculate()
-  local available_width = get_option'columns'
-  available_width = available_width - state.offset.left.width - state.offset.right.width
 
-  local pinned_count, pinned_sum, sum, widths = Layout.calculate_buffers_width()
+  local total_width = get_option('columns')
+
+  local left_width  = state.offset.left.width
+  local right_width = state.offset.right.width
   local tabpages_width = Layout.calculate_tabpages_width()
+  local buffers_width = total_width - state.offset.left.width - state.offset.right.width - tabpages_width
 
+  local pinned_count, pinned_sum, unpinned_sum, widths = Layout.calculate_buffers_width()
   local pinned_width = pinned_sum + (pinned_count * config.options.minimum_padding * SIDES_OF_BUFFER)
-  local buffers_width = available_width - pinned_width - tabpages_width
-  local count = #widths - pinned_count
 
-  local remaining_width = max(0, buffers_width - sum)
-  local remaining_width_per_buffer = floor(remaining_width / count)
+  local unpinned_allocated_width = buffers_width - pinned_width
+  local unpinned_count = #widths - pinned_count
+
+  local remaining_width = max(0, unpinned_allocated_width - unpinned_sum)
+  local remaining_width_per_buffer = floor(remaining_width / unpinned_count)
   local remaining_padding_per_buffer = floor(remaining_width_per_buffer / SIDES_OF_BUFFER)
-  local padding_width = max(config.options.minimum_padding, min(remaining_padding_per_buffer, config.options.maximum_padding))
-  local actual_width = sum + (count * padding_width * SIDES_OF_BUFFER)
-
-  return {
-    actual_width = actual_width,
-    base_widths = widths,
-    buffers_width = buffers_width,
-    padding_width = padding_width,
-    pinned_width = pinned_width,
-    scroll_max = max(0, actual_width - buffers_width),
-    tabpages_width = tabpages_width,
+  local padding = max(config.options.minimum_padding, min(remaining_padding_per_buffer, config.options.maximum_padding))
+
+  local unpinned_width = unpinned_sum + (unpinned_count * padding * SIDES_OF_BUFFER)
+
+  local buffers_used_width = pinned_width + unpinned_width
+
+  local result = {
+    total_width = total_width,
+
+    left = {
+      width = left_width,
+    },
+
+    buffers = {
+      width = buffers_width,
+      pinned_width = pinned_width,
+      unpinned_width = unpinned_width,
+      unpinned_allocated_width = unpinned_allocated_width,
+      used_width = buffers_used_width,
+      padding = padding,
+      base_widths = widths,
+      scroll_max = max(0, unpinned_width - unpinned_allocated_width),
+    },
+
+    tabpages = {
+      width = tabpages_width,
+    },
+
+    right = {
+      width = right_width,
+    },
   }
```

Comment:
`layout.*.width` now always gives the allocated width for that section. The other widths are the used widths except if named `*_allocated_width`.

Reply:
This is _much_ cleaner!